periodic/periodic.go

```diff
@@ -508,6 +514,9 @@ MainLoop:
 			}
 			targetElapsedDuration := time.Duration(int64(targetElapsedInSec * 1e9))
 			sleepDuration := targetElapsedDuration - elapsed
+			if r.RequestJitter {
+				sleepDuration += addJitter(sleepDuration)
```

Comment:
you probably want to call this in the initial warmup too

Reply:
I don't know if it's needed. The jitter is added continuously at every "sync" point, so the one added at the warmup time will make little or no difference in the entire run.

Reply:
You mentioned in the premise that envoy may have issue with thundering herd of simultaneous connections, it would seem to me the jitter is thus most important for the warmup / initial connection ?

Reply:
Sorry I wasn't clear in the write-up. The thundering herd problem is less related to the initial connect() calls (although this could also be problematic), but rather send() calls from all the parallel clients, every time they wake up.